### DISCUSSION TITLE
Getting this Istio-init error when policy applied for non root user

### DISCUSSION THREAD

**Discussion by smoharana1977** [istio/istio#53684] [General]
Hi, we have applied the policy to run the container in non root user.
Istio installation is up, however when we deploy the pod with sidecar enabled, deployment is throwing below error - 

     admission webhook "validation.gatekeeper.sh" denied the request:
        [azurepolicy-k8sazurev3allowedusersgroups-xxx]
        Container istio-init is attempting to run as disallowed user 0. Allowed
        runAsUser: {"ranges": [], "rule": "MustRunAsNonRoot"}

How to handle this? why istio-init required root access to initiate the proxy?

**Comment by ilrudie** [comment]
https://istio.io/latest/docs/setup/additional-setup/cni/

Using the CNI removes the need for privileged init containers.

**Reply by smoharana1977** [reply]
that means, after CNI installation, do I need disabled the sidecar injection from namespace?
Is there anyway to override the configuration to run the istio-init in non root user?

**Reply by howardjohn** [reply]
This removes the istio-init container but keeps the sidecar

**Comment by smoharana1977** [comment]
After installing the cni node agent, proxy has been configured with non root user access.
